### ISSUE TITLE
PE when domain starts with two underscores

### ISSUE THREAD

**Issue by Mika56** [Mika56/PHP-SPF-Check#24]
Original message from @Pascal76

```php
<?php
use Mika56\SPFCheck\SPFCheck;
use Mika56\SPFCheck\DNSRecordGetter;

require('vendor/autoload.php');

$checker = new SPFCheck(new DNSRecordGetter()); // Uses php's dns_get_record method for lookup.
var_dump($checker->isIPAllowed('172.82.227.233', 'm1.email.samsung.com'));
?>
```

=> string(2) "PE"

dig +short TXT m1.email.samsung.com |grep spf 
"v=spf1 redirect=__spf.campaign.adobe.com"

There is a problem for this domain begining with 2 "_" with php < 7

```php
                            if (version_compare(PHP_VERSION, '7', '>=')) {
                                if (!filter_var($domain, FILTER_VALIDATE_DOMAIN)) {
                                    return false;
                                }
                            } else {
                                /** @link http://stackoverflow.com/a/4694816/2898156 */
                                if (!(preg_match('/^([_a-z\d](-*[a-z\d])*)(\.([a-z\d](-*[a-z\d])*))*$/i', $domain) 
                                    && preg_match('/^.{1,253}$/', $domain)
                                    && preg_match('/^[^\.]{1,63}(\.[^\.]{1,63})*$/', $domain))
                                ) {
                                    return false;
                                }
                            }
```

**Comment by Mika56** [comment]
Hi,

Problem seems to be elsewhere. When using `DNSRecordGetter`, it uses PHP's `dns_get_record` which can only use UDP.
Adobe's SPF record is quite long and cannot fit in a single UDP packet, so PHP silently (...) fails and returns no TXT records.
You should use `DNSRecordGetterDirect` instead. The forth argument is `$udp`. Set it to `false` to force TCP:

```php
$checker = new SPFCheck(new DNSRecordGetterDirect('8.8.8.8', 53, 30, false));
var_dump($checker->isIPAllowed('172.82.227.233', 'm1.email.samsung.com')); # string(1) "+"
```

**Comment by Mika56** [comment]
Okay, so it took me some time and research. The UDP thing seems to be only on my computer, tough I have tried with both PHP 5 and 7.
I've only tested the fix I told you with PHP 7 because I misunderstood your message, sorry.
I've published a new fix in d5b0aa4fe98577751ccab9e6085f73bd28eccb75 which unifies the way host names are validated in both PHP 5 and PHP 7. It turns out PHP 7's `FILTER_VALIDATE_DOMAIN` does almost no tests (length, does not start with a dot, see php/php-src#826), so I just used an other regex that allows a host name to start with multiple underscores.
This is published as v1.1.2, available on Packagist
